modules/core/src/main/scala/org/tessellation/domain/snapshot/rewards/Rewards.scala

```diff
@@ -1,15 +1,102 @@
 package org.tessellation.domain.snapshot.rewards
 
 import cats.data.NonEmptySet
+import cats.effect.Async
+import cats.implicits.toComposeOps
+import cats.syntax.functor._
 
 import scala.collection.immutable.SortedSet
 
+import org.tessellation.dag.snapshot.SnapshotOrdinal
 import org.tessellation.schema.ID.Id
-import org.tessellation.schema.transaction.RewardTransaction
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{RewardTransaction, TransactionAmount}
+import org.tessellation.security.SecurityProvider
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric._
 
 object Rewards {
+  val epochs: PosInt = 4
+  val epochDurationInYears: PosDouble = 2.5
+  val epochRewards: PosDouble = 853333333.20
```

Comment:
I'm very much in favor of doing all arithmetics on integer numbers. So `Long` or `Amount` type. And always normalized. Doing arithmetic on floating-point's will lead to rounding errors which can accumulate to a significant amount over time.